src/Teams/Services/ManageTeamsService.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Threading.Tasks;
+using Teams.Data;
+using Teams.Models;
+using Teams.Security;
+using Teams.Repository;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System;
+
+namespace Teams.Services
+{
+    public class ManageTeamsService : IManageTeamsService
+    {
+        private readonly ICurrentUser _currentUser;
+        private readonly IRepository<Team,int> _teamRepository;
+        public ManageTeamsService(ICurrentUser currentUser, IRepository<Team, int> teamRepository)
+        {
+            _currentUser = currentUser;
+            _teamRepository = teamRepository;
+        }
+
+        public async Task<bool> AddTeamAsync(string teamName)
+        {
+            if (Regex.IsMatch(teamName, ("^[a-zA-Z0-9-_,.]+$")) && !_teamRepository.GetAll().Any(t => t.TeamName.ToUpper().Equals(teamName.ToUpper())))
+            {
+                await _teamRepository.InsertAsync(new Team { TeamOwner = _currentUser.Current.Id(), TeamName = teamName });
```

Comment:
InsertAsync as far as I remember returns count of affected rows, you can use check if the result is greater than 0, then the insert is successful

Reply:
returns bool